zc_common/remote_resource/renderers.py

```diff
@@ -0,0 +1,323 @@
+    """
+
+    @classmethod
+    def extract_included(cls, request, fields, resource, resource_instance, included_resources):
+        # this function may be called with an empty record (example: Browsable Interface)
+        if not resource_instance:
+            return
+
+        included_data = list()
+        current_serializer = fields.serializer
+        context = current_serializer.context
+        included_serializers = utils.get_included_serializers(current_serializer)
+        included_resources = copy.copy(included_resources)
+        included_resources = [inflection.underscore(value) for value in included_resources]
+
+        for field_name, field in six.iteritems(fields):
+            # Skip URL field
+            if field_name == api_settings.URL_FIELD_NAME:
+                continue
+
+            # Skip fields without relations or serialized data
+            if not isinstance(field, (relations.RelatedField, relations.ManyRelatedField, BaseSerializer)):
+                continue
+
+            try:
+                included_resources.remove(field_name)
+            except ValueError:
+                # Skip fields not in requested included resources
+                # If no child field, directly continue with the next field
+                if field_name not in [node.split('.')[0] for node in included_resources]:
+                    continue
+
+            try:
+                relation_instance = getattr(resource_instance, field_name)
+            except AttributeError:
+                try:
+                    # For ManyRelatedFields if `related_name` is not set we need to access `foo_set` from `source`
+                    relation_instance = getattr(resource_instance, field.child_relation.source)
+                except AttributeError:
+                    if not hasattr(current_serializer, field.source):
+                        continue
+                    serializer_method = getattr(current_serializer, field.source)
+                    relation_instance = serializer_method(resource_instance)
+
+            if isinstance(relation_instance, Manager):
+                relation_instance = relation_instance.all()
+
+            new_included_resources = [key.replace('%s.' % field_name, '', 1)
+                                      for key in included_resources
+                                      if field_name == key.split('.')[0]]
+            serializer_data = resource.get(field_name)
+
+            if isinstance(field, RemoteResourceField):
+                user_id = getattr(request.user, 'id', None)
+                roles = request.user.roles
+                pk = serializer_data.get('id')
+
+                include = ",".join(new_included_resources)
+                try:
+                    remote_resource = event_client.get_remote_resource_data(
+                        field_name, pk=pk, user_id=user_id,
+                        include=include, page_size=1000, roles=roles)
```

Comment:
It's fine for now, just wanted to make sure the most possible. We can change that later.